Add back keyboard shortcut features (#9)

muachilin opened the pull request:
Add back previously-supported keyboard shortcut features.

muachilin committed:
Add back keyboard shortcut features

krisgun commented:
My guess is that it's going to take a bit more work than just restoring the file from the old version of the project. The current GUI library that is used is JavaFX and just looking at the imports in EmacsKeyBindings there are a lot of Swing imports, so there would probably have to be some porting done

muachilin commented:
I see! I will look into handling the new GUI framework. Thanks!

krisgun committed:
[feat] Add gui support

This commit adds a new preference under the Entry editor tab in the preferences. This also maps to the preferences:

EDITOR_EMACS_KEYBINDINGS
EDITOR_EMACS_KEYBINDINGS_REBIND_CA
EDITOR_EMACS_KEYBINDINGS_REBIND_CF

[Issue: #9]

krisgun committed:
[fix] Added missing text strings

The text strings describing the emacs options in preferences were missing, and are now added.

[Issue: #9]

Zodbigt committed:
[feat] Keyboard shortcuts for data entry/editing
Ths commit adds the first step for re-integrating emac keybindings.
In this version emacs works globly but dosent have all features yet.

[issue: #9]

Zodbigt committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

krisgun committed:
[feat] Emacs key bindings can now be toggled in preferences

Emacs key bindings can now be toggled through a preference option in the Entry Editor preference tab

[Issue: #9]

krisgun committed:
[feat] Ctrl+a and Ctrl+f can now also be toggled in preferences

There are sub-options deciding whether C-a and C-f  should be rebinded as well, which now work as intended

[Issue: #9]

muachilin committed:
Add more Emacs features back

muachilin committed:
[fix] Modify the ordering of imports

krisgun committed:
[fix] Change import order and add missing localization keys

Imports were not conforming to the CheckStyle and some localization keys for the Emacs commands were missing

[Issue: #9]

krisgun committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

krisgun committed:
[fix] Reorder import and add localization keys

[Issue: #9]

muachilin committed:
[feat] Add shortcuts for upcasing and lowercasing words

muachilin committed:
[fix] Add whitespaces after if statement

Zodbigt committed:
[feat] Moves the Emacs lgic to is own class

This moves all the emacs moves to its own class.

[issue: #9]

Zodbigt committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

krisgun committed:
[fix] Rename EmacsKeyBindings and reorder imports

The old EmacsKeyBindings class is now removed and replaced with a working one. Missing localization keys were also added.

[Issue: #9]

muachilin committed:
[fix] Modify the capitalization feature

krisgun committed:
[fix] Checkboxes are now disabled when parent is unchecked

If the enabledEmacsKeyBindings checkbox is unchecked, its children will now be disabled to further convey that the children settings are not applied anymore.

[Issue: #9 ]

krisgun committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

muachilin committed:
[feat] Refactor the util function of editting next word

muachilin committed:
[feat] Refactor the util function of editting next word

muachilin committed:
[feat] Refactor the util function of editting next word

muachilin committed:
[fix] Add empty line between functions

muachilin committed:
[feat] Add feature of deleting whole line

muachilin committed:
[feat] Add feature of deleting the next word and refactor util functions

muachilin committed:
[feat] Refactor util functions

Zodbigt committed:
[feat] Add test for the StringChangeNextWord class

Thois commit adds a start for tests for the StringChangeNextWord class.

[issue: #9]

muachilin committed:
[feat] Add feature of deleting the previous word

muachilin committed:
[feat] Add feature of deleting the previous word

Zodbigt committed:
[feat] Add test for the StringChangeNextWord class

This commit adds more test case for the StringChangeNextWord class.

[issue: #9]

Zodbigt committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

krisgun committed:
[fix] StringChangeNextWord trailing space now removed

[Issue: #9]

stevensdavid committed:
Make methods static

stevensdavid committed:
Use StringBuilder and fix style

muachilin committed:
[fix] Modify some features

krisgun committed:
[fix] Change parameter order in assertEquals

[Issue: #9]

krisgun committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

muachilin committed:
[fix] Modify some features

muachilin committed:
[fix] Modify some features

Zodbigt committed:
[fix] Add comments for postion of the caret in StringChangeNextWordTest class

[issue: #9]

Zodbigt committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

krisgun committed:
[fix] Emacs key bindings can now be used in all classes that inherit TextInputControl

[Issue: #9]

krisgun committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

Zodbigt committed:
[feat] Add test cases for the StringChangeNextWordTest class

[issue: #9]

stevensdavid committed:
Refactor case changing

stevensdavid committed:
Merge branch 'issue/9' of github.com:stevensdavid/jabref into issue/9

Zodbigt committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

stevensdavid committed:
Merge branch 'issue/9' of github.com:stevensdavid/jabref into issue/9

muachilin committed:
[fix] Modify deleting next word

muachilin committed:
[fix] Modify deleting next word

stevensdavid committed:
Fix refactor

stevensdavid committed:
Merge branch 'issue/9' of github.com:stevensdavid/jabref into issue/9

muachilin committed:
[feat] Add feature of deleting previous word

muachilin committed:
[feat] Add feature of deleting previous word

muachilin committed:
[fix] Remove unused numOfSpace

krisgun committed:
[refac] Moved more of the Emacs logic into its class

A lot of preferences were passed from JabRefGUI instead of getting them directly from the preferences.

[Issue: #9]

stevensdavid committed:
Fix refactor and tests

stevensdavid committed:
Merge branch 'issue/9' of github.com:stevensdavid/jabref into issue/9

Zodbigt committed:
[feat] Add new class for returning both caret postion and result text

This commit adds a new class to handle borth caret postion and result text for the emacs comands.

[issue: #9]

Zodbigt committed:
Merge branch 'issue/9' of https://github.com/stevensdavid/jabref into issue/9

stevensdavid committed:
Merge branch 'issue/9' of github.com:stevensdavid/jabref into issue/9

stevensdavid committed:
Fix outofbounds

stevensdavid committed:
Rename classes and methods

stevensdavid committed:
Fix typo

krisgun committed:
[fix] Update localization keys

[Issue: #9]
